tap_github/client.py

```diff
@@ -0,0 +1,271 @@
+        Call rest API to verify that the user has sufficient permissions to access this repository.
+        """
+        try:
+            self.authed_get("verifying repository access", url_for_repo)
+            LOGGER.info("Verifying access of repository: %s", repo)
+        except NotFoundException:
+            # Throwing user-friendly error message as it checks token access
+            message = "HTTP-error-code: 404, Error: Please check the repository name \'{}\' or you do not have sufficient permissions to access this repository.".format(repo)
+            raise NotFoundException(message) from None
+
+    def verify_access_for_repo(self):
+        """
+        For all the repositories mentioned in the config, check the access for each repos.
+        """
+        access_token = self.config['access_token']
+        self.session.headers.update({'authorization': 'token ' + access_token, 'per_page': '1', 'page': '1'})
+
+        repositories = self.extract_repos_from_config()
+
+        for repo in repositories:
+
+            url_for_repo = "https://api.github.com/repos/{}/commits".format(repo)
+
+            # Verifying for Repo access
+            self.verify_repo_access(url_for_repo, repo)
+
+    def extract_repos_from_config(self):
+        """
+        Extracts all repositories from the config and calls get_all_repos()
+        for organizations using the wildcard 'org/*' format.
+        """
+        repo_paths = list(filter(None, self.config['repository'].split(' ')))
+
+        orgs_with_all_repos = list(filter(lambda x: x.split('/')[1] == '*', repo_paths))
+
+        if orgs_with_all_repos:
+            # Remove any wildcard "org/*" occurrences from `repo_paths`
+            repo_paths = list(set(repo_paths).difference(set(orgs_with_all_repos)))
+
+            # Get all repositories for an org in the config
+            all_repos = self.get_all_repos(orgs_with_all_repos)
+
+            # Update repo_paths
+            repo_paths.extend(all_repos)
+
+        return repo_paths
+
+    def get_all_repos(self, organizations: list):
+        """
+        Retrieves all repositories for the provided organizations and
+        verifies basic access for them.
+
+        Docs: https://docs.github.com/en/rest/reference/repos#list-organization-repositories
+        """
+        repos = []
+
+        for org_path in organizations:
+            org = org_path.split('/')[0]
+            for response in self.authed_get_all_pages(
+                'get_all_repos',
+                'https://api.github.com/orgs/{}/repos?sort=created&direction=desc'.format(org)
+            ):
+                org_repos = response.json()
+
+                for repo in org_repos:
+                    repo_full_name = repo.get('full_name')
+
+                    LOGGER.info("Verifying access of repository: %s", repo_full_name)
+                    self.verify_repo_access(
+                        'https://api.github.com/repos/{}/commits'.format(repo_full_name),
```

Comment:
We are using this URL twice, is it possible to define this in one variable?

Reply:
Defined URL in single variable `base_url`.